src/_pytest/reports.py

```diff
@@ -264,6 +265,8 @@ def from_item_and_call(cls, item, call):
         Factory method to create and fill a TestReport with standard item and call info.
         """
         when = call.when
+        if not call.stop:
```

Comment:
in my opinion this correction should never be necessary, initially a call-info was complete after the creation

Reply:
hmm,actually scrap that, with the  overall structural flow its sensible

its a necessary action at a distance in the new flow

i believe this is indicative of why at some perspective i would prefer a call-info that's more static in nature, but unit-test isn't exactly our friend there